### PR TITLE
[Docs][8.16] Link to ECK Azure snapshot docs

### PR DESCRIPTION
**Pull request by leemthompo**
### ℹ️ Can't merge until ECK 2.15 goes live, see [comment](https://github.com/elastic/cloud-on-k8s/pull/7980#issuecomment-2270489083)

Links to: https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-snapshots.html#k8s-azure-workload-identity

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
Easy when you know how :) LGTM

File: docs/reference/snapshot-restore/repository-azure.asciidoc [hunk 1]
```diff
@@ -182,7 +182,7 @@ is running.
 
 When running {es} in
 https://azure.microsoft.com/en-gb/products/kubernetes-service[Azure Kubernetes
-Service], for instance using {eck-ref}[{eck}], you should use
+Service], for instance using {eck-ref}/k8s-snapshots.html#k8s-azure-workload-identity[{eck}], you should use
```
Comment (leemthompo) [inline docs/reference/snapshot-restore/repository-azure.asciidoc:185]:
```suggestion
Service], for instance using https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-snapshots.html#k8s-basic-snapshot-azure[{eck}], you should use
```

Comment (DaveCTurner) [reply docs/reference/snapshot-restore/repository-azure.asciidoc:185]: Hm that's not linking to the right subsection tho, https://github.com/elastic/cloud-on-k8s/pull/7980 removed `k8s-basic-snapshot-azure`.

Comment (leemthompo) [reply docs/reference/snapshot-restore/repository-azure.asciidoc:185]: yeah just deleted comment, didn't realize you had commented on it 😆 . this just broke my skull, do we have to wait for k8s master docs to flip to whatever the next current switchover is?

**Comment by leemthompo** [comment]
run docs-build

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/116835)|
|✅|8.16|[](https://github.com/elastic/elasticsearch/pull/116836)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**